Sources/SourceControl/GitRepository.swift

```diff
@@ -85,6 +85,8 @@ public struct GitRepositoryProvider: RepositoryProvider, Cancellable {
     private let cancellator: Cancellator
     private let git: GitShellHelper
 
+    private var repositoryCache = ThreadSafeKeyValueStore<String, Repository>()
```

Comment:
nit: maybe `repositoryHandleCache` or something? Just because otherwise it can be mistaken to be related to the on-disk repository by a casual reader.